x/examples/fetch/main.go

```diff
@@ -89,12 +102,27 @@ func main() {
 		}
 		return dialer.DialStream(ctx, net.JoinHostPort(host, port))
 	}
-	httpClient := &http.Client{Transport: &http.Transport{DialContext: dialContext}, Timeout: time.Duration(*timeoutSecFlag) * time.Second}
+	httpClient := &http.Client{
+		Transport: &http.Transport{DialContext: dialContext},
+		Timeout:   time.Duration(*timeoutSecFlag) * time.Second,
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
 
 	req, err := http.NewRequest(*methodFlag, url, nil)
 	if err != nil {
 		log.Fatalln("Failed to create request:", err)
 	}
+	for _, headerLine := range headersFlag {
+		name, value, _ := strings.Cut(headerLine, ":")
+		name = strings.TrimSpace(name)
+		value = strings.TrimSpace(value)
+		if name == "" || value == "" {
```

Comment:
[IETF allows header value to contain only spaces (LWS, linear white space)](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2), maybe we just need to check `name`?

Reply:
It doesn't, actually
<img width="779" alt="image" src="https://github.com/Jigsaw-Code/outline-sdk/assets/113565/1273853d-90a2-433d-93f8-17893c77149e">

Reply:
Oh, actually it can have space if it's quoted!